### PR TITLE
Changes citation/thesis download filename

### PR DESCRIPTION
**Pull request by bryjbrown**
**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-1805

# What does this Pull Request do?
The current version of Scholar formats download links for citation/thesis cmodels as /islandora/object/PID/datastream/PDF/download/citation.pdf. This could lead to a lot of "citation.pdf" files (or "citation ($).pdf" if your computer automatically renames duplicate filenames) in your downloads folder if you download files from multiple theses/citations.

This PR changes the formatted download link to be "PID.pdf" with the colon swapped to an underscore. This way, if people download PDFs from multiple objects, they can at least keep them apart.

I considered formatting the file name as the object label, but there are too many potential special characters that could lead to weird behavior when used in a filename. This seems safer.

# How should this be tested?
Boot up a VM with this PR and check out a citation/thesis object's "PDF" download link. The link should end in "PID.pdf" instead of "citation.pdf". Download the file to your local machine and check the filename.

# Interested parties
@DonRichards @rosiel @DiegoPino @bondjimbond

### CODE REVIEW

**Comment by bryjbrown** [comment]
Pinging @Islandora/7-x-1-x-committers

**Review by DiegoPino — APPROVED** [review]
Looks good. Not sure if a PID makes sense for anyone outside islandora world but much better than citation

**Comment by bondjimbond** [comment]
This is a useful and valuable change! @bryjbrown ++

**Comment by bryjbrown** [comment]
@DiegoPino Yeah, originally I was thinking of formatting the title or using the object label as the file name, but there can be a lot of special characters in a title that wouldn't work in a filename. I'm not about to take on responsibility for learning what is a valid filename across different operating systems, so PIDs it is! :smile_cat:

**Comment by rosiel** [comment]
It's approved? i'll hit the merge button (sorry for the delay; cleaning out my inbox)

**Comment by DiegoPino** [comment]
@rosiel quick though, at least for this release, would you be in favor of something like "who approved (tested) merges if can"?  I keep coming to merge stuff I have on my calendar to be merged (after I approved it) just to find it merged. Not a biggie really, it is just an idea, since we are all a bit short of time. My main concern comes really from merging something not tested by the merger but by the approver, but again, trusting is good of course! it is just that sometimes when things fail the merger is to be blamed! and nobody ever remembers who tested/approved. Anyway, just an idea

**Comment by rosiel** [comment]
I'd say we put it to the committers to a vote. (and not here, on the private committers mailing list).

You might be the only one who schedules merges on their calendar, and I might be the only one who gets behind on their emails because every pull request is a minimum commitment of 15 minutes to read to fully understand the context, which I feel obliged to do to every one but especially if I've  been personally tagged in it, and when I can't act on it when it's clearly waiting for something to happen I have wasted my time, and that's frustrating. If we trust the reviewer to have reviewed it and have looked at the code changes (like in this case) and have run it (I know I didn't have to, but I had in this case), why can't we hit merge undertaking the risk that we'll get blamed if anything goes awry? 

Because if we have it your way, then I have to decide, "Should I do nothing?" as well as "am I an annoying bitch for sending a reminder to the reviewer to hit merge?" And that depends on who it is, how long the PR's been standing, whether there were weekends/holidays involved, whether the reviewer is allowed to work on 7.x during work time, etc. And that's a way harder (and riskier) decision for me to make (again, speaking only for myself here) than whether to push code (that I trust) which someone else has approved (who I trust).

Unless you're taking stats? Do other committers want to see their name as the merger on the commit logs?

**Comment by bryjbrown** [comment]
@rosiel++ for discussing with committers. This seems like a larger issue than what should be discussed in a thread on a merged issue :+1:

**Comment by dannylamb** [comment]
I thought we hashed this out at a recent call and said it would be ok for @rosiel to merge in this scenario?  We really should take notes...

Nonetheless, there's nothing wrong with adding this as an agenda item if we feel the need to re-discuss.

**Comment by DiegoPino** [comment]
@dannylamb can't recall anything specific to @rosiel and she being able to merge on certain scenarios on the last call, maybe i missed a bit there, but yeah, you are probably right. I can't even recall someone telling anyone that is a committer it should/could not merge in any call ever. Kinda confused, i remember mentioning that mergers should test but I guess I need to apologize for asking for that.

We don't have any rule that supports my request or whatever, but also we lack mentions to github's approval/change request -> merge workflow and sadly the /reminder stuff does not work still. I still try to follow the rule of respecting other peoples work and use common sense in the absence of written rules, probably failing there also.

@rosiel even so, with all my faults, I don't understand your reaction/response here to an idea, meaning well and respectfully formulated, I actually feel uncomfortable with it, so for sake of sanity  (yours and mine) dismiss the request and do as I never posted that. My apologies

Thanks

**Comment by rosiel** [comment]
If giving credit is desired, a friend of mine just posted about the ability to add co-committers to a commit. Would that be of interest? https://help.github.com/articles/creating-a-commit-with-multiple-authors/

**Comment by dannylamb** [comment]
@DiegoPino It was nothing specific to this PR or to you or @rosiel .  The discussion I'm recalling was about who can merge once approved and 24 hours have passed.  But this is a perfect example of oral history dictating workflow, which shouldn't happen.  The workflow should be clear enough and if it's not we should update it.

**Pull request merged**
